oteapi_dlite/strategies/mapping.py

```diff
@@ -242,15 +242,27 @@ class URIs within a specified RDF graph.
         {% macro sparql_query(class_names, graph_uri) %}
         PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
         PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+        PREFIX owl: <http://www.w3.org/2002/07/owl#>
+
         SELECT ?parentClass
         WHERE {
-          GRAPH <{{ graph_uri }}> {
-            ?class rdfs:subClassOf* ?parentClass .
-            FILTER(
-                {% for class_name in class_names -%}
-                    ?class = <{{ class_name }}>{{ " ||" if not loop.last }}
-                {% endfor %})
-          }
+            GRAPH <{{ graph_uri }}> {
+                {
+                # Subclass traversal to find parent classes
+                ?class rdfs:subClassOf* ?parentClass .
+                FILTER(
+                    {% for class_name in class_names -%}
+                        ?class = <{{ class_name }}>{{ " ||" if not loop.last }}
+                    {% endfor %}
+                )
+                }
+            UNION
+                {
+                # Explicitly include owl:Thing as a parent class
+                VALUES ?class { {% for class_name in class_names -%} <{{ class_name }}> {% endfor %} }
```

Comment:
I don't see how this fixes anything. You set `Thing` as parent for all parentClasses?!. I would rather suggest that you leave the original query, but in the code that calls it, if the query doesn't return a single parent class, call a different method that fetches the entire graph.

Reply:
> I don't see how this fixes anything. You set `Thing` as parent for all parentClasses?!. I would rather suggest that you leave the original query, **but in the code that calls it, if the query doesn't return a single parent class, call a different method that fetches the entire graph.**

separate method sounds good